quickstart/docker/image/ziti-cli-functions.sh

```diff
@@ -525,6 +559,180 @@ function ziti_expressConfiguration {
   echo ""
 }
 
+function checkEdgeRouterENV {
+  if [[ "${ZITI_CTRL_ADVERTISED_ADDRESS-}" == "" ]]; then
+    echo ""
+    echo -e "$(RED "ERROR: Must declare: ZITI_CTRL_ADVERTISED_ADDRESS")"
+    exit
+  else
+    echo -e "Use $(BLUE "${ZITI_CTRL_ADVERTISED_ADDRESS}") as Controller address"
+  fi
+
+  if [[ "${ZITI_CTRL_PORT-}" == "" ]]; then
+    echo ""
+    echo -e "$(RED "ERROR: Must declare: ZITI_CTRL_PORT")"
+    exit
+  else
+    echo -e "Use $(BLUE "${ZITI_CTRL_PORT}") as Controller port"
+  fi
+
+  if [[ "${ZITI_EDGE_CONTROLLER_PORT-}" == "" ]]; then
+    echo ""
+    echo -e "$(RED "ERROR: Must declare: ZITI_EDGE_CONTROLLER_PORT")"
+    exit
+  else
+    echo -e "Use $(BLUE "${ZITI_EDGE_CONTROLLER_PORT}") as Controller API port"
+  fi
+
+  if [[ "${ZITI_EDGE_ROUTER_IP_OVERRIDE-}" == "" ]]; then
+    echo ""
+    echo -e "$(RED "ERROR: Must declare: ZITI_EDGE_ROUTER_IP_OVERRIDE")"
+    exit
+  else
+    echo -e "Use $(BLUE "${ZITI_EDGE_ROUTER_IP_OVERRIDE}") as Edge Router Advertise Address"
+  fi
+
+  if [[ "${ZITI_EDGE_ROUTER_PORT-}" == "" ]]; then
+    echo ""
+    echo -e "$(RED "ERROR: Must declare: ZITI_EDGE_ROUTER_PORT")"
+    exit
+  else
+    echo -e "Use $(BLUE "${ZITI_EDGE_ROUTER_PORT}") as Edge Router Advertise Port"
+  fi
+
+  # find the interface, the interface is not been used right now.
+  # it can be used later on if we created ER tunneler part in tproxy mode
+  main_interface=$(ip route get 8.8.8.8 | awk -- '{printf $5}')
+  echo -e "Use $(BLUE "${main_interface}") as main interface"
+
+  echo ""
+  echo "EdgeRouter Environment variable processed correctly."
+}
+
+# dump router environment variable for debugging
+function dump_router_env {
+  echo -e "Controller Address and Port: ${ZITI_CTRL_ADVERTISED_ADDRESS}:${ZITI_CTRL_PORT}" 
+  echo -e "        Controller API port: ${ZITI_EDGE_CONTROLLER_PORT}"
+  echo -e "    Edge Router IP and Port: ${ZITI_EDGE_ROUTER_IP_OVERRIDE}:${ZITI_EDGE_ROUTER_PORT}"
+  echo -e "                  Interface: ${main_interface}"
+  echo -e "                  ZITI-HOME: ${ZITI_HOME}"
+  echo -e "       ZITI Binary location: ${ZITI_BIN_DIR-}"
+  echo -e "                router-name: ${ername}"
+  echo -e ""
+}
+
+function ziti_expressRouter {
```

Comment:
This feels like going through the quickstart process again, I mentioned this in a main PR comment but would it be possible to modify `createEdgeRouterConfig` to create the router config rather than a new express install style function, I'm worried that with the reused variables and duplicated code is going to introduce a lot more points of failure.

Reply:
it was make to be similar to quickstart minus the creation of the controller.

the idea is the directory structures and etc. are similar to quickstart, so we can find config/env quickly.  It is also not lump into quickstart, in case there is bug, we can fix them easier.  At some point, we can try to consolidate functions if we choose to.

Reply:
The quickstart does many different things, downloading ziti, setting up a PKI, creating a controller, router, etc. I don't think all of this is necessary for just creating a router.

With this being essentially copied code, if there is a bug that we find we'll have to remember to fix it in both places. In addition, even just making an enhancement, we'll have to review all of this code too.

For instance, this entire first if statement is a copy of what's in express install. If we were going to use this in two places it should definitely be in a separate function and just referenced by each place calling that function. However, I'm not suggesting we do that as I don't think this first if statement, which checks if expressInstall was run, pertains to someone wanting to add a single new router. This would clear all of their env variables and they'd have to later re-source the main quickstart `.env` file which seems like a step backward.

Reply:
this code  is adding router outside of the vm. so there should be one env file.